services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -72,380 +72,749 @@ class HakemusServiceITest(
 ) : IntegrationTest() {
 
     @Nested
-    inner class HakemusResponse {
-        @Test
-        fun `when application does not exist should throw`() {
-            assertThat(applicationRepository.findAll()).isEmpty()
+    inner class Common {
 
-            val exception = assertFailure { hakemusService.hakemusResponse(1234) }
+        @Nested
+        inner class HakemusResponse {
```

Comment:
I think having all the tests for a single method in the same place is better. So, like:

```
@Nested
inner class HakemusResponse {
    @Test
    fun `when application does not exist should throw`() {
    ...

    @Nested
    inner class WithJohtoselvityshakemus {
        @Test
        fun `returns yhteystiedot and yhteyshenkilot if they're present`() {
    ...
    }

    @Nested
    inner class WithKaivuilmoitus {
        @Test
        fun `returns yhteystiedot and yhteyshenkilot if they're present`() {
    ...
    }
}
```

Reply:
Refactored as suggested.